chainer/training/extensions/sigmoid_shift.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import division
+
+import numpy
+
+from chainer.training import extension
+
+
+class SigmoidShift(extension.Extension):
+
+    """Trainer extension to sigmoid shift an optimizer attribute.
+
+    This extension decreases the specified attribute of
+    an optimizer using the sigmoid function. This extension will
+    set the corresponding attribute to:
+    ``init_value * e^{-1 * rate * i} / (e^{-1 * rate * i} + 1)``,
+    at the ``i``-th iteration, where rate is the rate of shift.
+
+    This extension is also called before the training loop starts
+    by default.
+
+    Args:
+        attr (str): Name of the attribute to shift.
+        rate (float): Rate of sigmoid shift.
+        init (float): Initial value of the attribute. If it is ``None``, the
+            extension extracts the attribute at the first call and uses it as
+            the initial value.
+        target (float): Target value of the attribute. If the attribute reaches
+            this value, the shift stops.
+        optimizer (~chainer.Optimizer): Target optimizer to adjust the
+            attribute. If it is ``None``, the main optimizer of the updater is
+            used.
+    """
```

Comment:
Could you cite a reference for the method?

Reply:
I don't believe this method is used specifically in any papers, and is therefore experimental.